src/OpenTracing/ScopeManager.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace OpenTracing;
+
+/**
+ * Keeps track of the current active `Span`.
+ */
+interface ScopeManager
```

Comment:
will there be a shared implementation of this interface?

Reply:
Not sure. I'd like to get some feedback from @felixfbecker or @beberlei to see if it makes sense for this to be an interface or to ship objects.

Reply:
@yurishkuro what do you mean by shared?

Reply:
Ah you mean an implementation that libraries can use, because they will implement the same code anyways.

Maybe we take a look at the first implementations and decide at a later point?

Reply:
in Java the opentracing API provides a default implementation, ThreadLocalScopeManager. @jcchavezs said php doesn't really use threads, so a simple singleton implementation with a single stack of Scopes would work for most cases (unless one uses event-loop-like frameworks).

Reply:
(disclaimer: you can use threads in PHP with pthreads but only if not a SAPI, I am pretty sure 99% of the cases don't use threads in PHP)

Reply:
So this is the interface one would have to implement to make it work in an event loop? Could you explain a bit more what that vs the normal implementation would look like?

Reply:
Also, the user base for event loops in PHP is growing - take react/event-loop install statistics as an example: https://packagist.org/packages/react/event-loop/stats

Reply:
@felixfbecker growing yes, more than 1% deployment base? probably not. I agree about a shared implementation that assumes "Shared Nothing Single Request per Process" what PHP was designed for.